api/tests/opentrons/protocol_engine/test_types.py

```diff
@@ -20,3 +21,17 @@ def test_handles_invalid_hex(invalid_hex_color: str) -> None:
         HexColor(invalid_hex_color)
     with pytest.raises(ValidationError):
         HexColor.model_validate_json(f'"{invalid_hex_color}"')
+
+
+class TestModel(BaseModel):
+    """Test model for deserializing SimulatedProbeResults."""
+
+    value: float | SimulatedProbeResult
+
+
+def test_deserializes_simulated_liquid_probe() -> None:
+    """Should be able to roundtrip our simulated results."""
+    base = TestModel(value=SimulatedProbeResult())
+    serialized = base.model_dump_json()
+    deserialized = TestModel.model_validate_json(serialized)
+    assert isinstance(deserialized.value, SimulatedProbeResult)
```

Comment:
To sate my paranoia, can we expand this test to cover the case where the JSON has a value that doesn't represent a `SimulatedProbeResult`?

Something like:

```suggestion


def test_simulated_probe_result_serdes() -> None:
    """Should be able to roundtrip our simulated results."""

    class TestModel(BaseModel):
        """Test model for deserializing SimulatedProbeResults."""

        value: float | SimulatedProbeResult

    # Should be able to roundtrip our simulated results:
    base = TestModel(value=SimulatedProbeResult())
    serialized = base.model_dump_json()
    deserialized = TestModel.model_validate_json(serialized)
    assert isinstance(deserialized.value, SimulatedProbeResult)

    # Should allow parsing to fall back to other union elements (in this case a float):
    assert TestModel.model_validate_json('{"value": 1.23}') == TestModel(value=1.23)

    # Parsing should totally error out if the value doesn't fit into any of the union elements:
    with pytest.raises(ValidationError):
        TestModel.model_validate_json(
            '{"value": "not a valid float or SimulatedProbeResult"}'
        )

```